backend/modules/query_engines/default_engine.py

```diff
@@ -0,0 +1,71 @@
+from fastapi import HTTPException
+from langchain.chains import RetrievalQA
+from langchain.chains.question_answering import load_qa_chain
+from langchain.prompts import PromptTemplate
+from langchain.schema.vectorstore import VectorStoreRetriever
+
+from backend.modules.query_engines.base import BaseQueryEngine
+from backend.utils.base import DefaultQueryInput
+from backend.utils.decorator import Post, QueryEngine
+from backend.utils.logger import logger
+
+
+@QueryEngine("/default")
+class DefaultQueryEngine(BaseQueryEngine):
+    """
+    Default Query Engine
+    uses langchain retrieval qa to answer the query
+    """
+
+    @Post("/query")
+    def query(self, request: DefaultQueryInput):
+        try:
+            # Get the vector store
+            vector_store = self._get_vector_store(request.collection_name)
+
+            # Get the LLM
+            llm = self._get_llm(request.model_configuration)
+
+            # Create the retriever using langchain VectorStoreRetriever
+            retriever = VectorStoreRetriever(
+                vectorstore=vector_store,
+                search_type=request.retriever_config.get_search_type,
+                search_kwargs=request.retriever_config.get_search_kwargs,
+            )
+            DOCUMENT_PROMPT = PromptTemplate(
+                input_variables=["page_content"],
+                template="<document>{page_content}</document>",
+            )
+            QA_PROMPT = PromptTemplate(
+                input_variables=["context", "question"],
+                template=request.prompt_template,
+            )
+
+            # Create the QA chain
+            qa = RetrievalQA(
+                retriever=retriever,
+                llm=llm,
+                combine_documents_chain=load_qa_chain(
+                    llm=llm,
+                    chain_type="stuff",
+                    prompt=QA_PROMPT,
+                    document_variable_name="context",
+                    document_prompt=DOCUMENT_PROMPT,
+                    verbose=True,
+                ),
+                return_source_documents=True,
+                verbose=True,
+            )
+
+            # Get the answer
+            outputs = qa({"query": request.query})
+
+            return {
+                "answer": outputs["result"],
+                "docs": outputs.get("source_documents") or [],
+            }
+        except HTTPException as exp:
```

Comment:
Not sure why this would be raised, this is meant to be raised by us
maybe we mean exceptions from httpx or requests?

Reply:
yes!

Reply:
This will never be thrown, right?
you need to import the relevant exception from httpx, requests
Although I don't see the point of catching and re-raising
lines 68-70 will have same effect